virtcontainers/clh.go

```diff
@@ -138,7 +138,7 @@ func (clh *cloudHypervisor) createSandbox(ctx context.Context, id string, networ
 			return err
 		}
 
-		if clh.version.Major < supportedMajorVersion && clh.version.Minor < supportedMinorVersion {
+		if clh.version.Major < supportedMajorVersion || (clh.version.Major == supportedMajorVersion && clh.version.Minor < supportedMinorVersion) {
```

Comment:
Ideally this change would be coupled with:

- A refactor so this logic is encapsulated into a function (something like `func (clh *cloudHypervisor) checkVersion()`).
- A table-based test exercising boundary cases for all the version variables used here (https://github.com/kata-containers/tests/blob/master/Unit-Test-Advice.md#table-driven-tests).

However, there was talk of this test being temporary (https://github.com/kata-containers/runtime/pull/2164/files#r345596109), so it's unclear to me how long this test will remain in the code? If it's "a while", I vote we add some tests :smile:

/cc @sameo, @ericooper, @jcvenegas.

Reply:
I actually quite disagree. I in fact did not add this version check temporarily. the hypervisor binary is not part of the source tree so can be anything and it's cli syntax is still under constant revision which is fine. I did not add the builder pattern without a reason. In future versions of the hypervisor different builders can be instantiated based on that version to accommodate evolution of the cloud hypervisor cli. In production we see very often that systems break causing hours of troubleshooting because of these kind of implicit assumptions. This version check should help to avoid this. And although the current logic perhaps does not make perfectly sense it is there for future usage.

Reply:
Thanks for clarifying @ericooper. In which case, we definitely need unit tests for this imho.

Reply:
@jodh-intel 
I have added test for the version check.

Please take a look.